src/plexil-adapter/OwInterface.cpp

```diff
@@ -237,88 +237,44 @@ static void handle_joint_fault (Joint joint, int joint_index,
   handle_overtorque (joint, msg->effort[joint_index]);
 }
 
-void OwInterface::systemFaultMessageCallback
-(const  ow_faults::SystemFaults::ConstPtr& msg)
+template <typename T1, typename T2>
+void OwInterface::faultCallback (T1 msg_val, T2& fmap,
+                                 const string& component)
 {
-  // Publish all joint information for visibility to PLEXIL and handle any
-  // system-level fault messages.
-  uint64_t msg_val = msg->value;
-
-  for (auto const& entry : systemErrors){
+  for (auto const& entry : fmap) {
     string key = entry.first;
-    uint64_t value = entry.second.first;
-    bool b = entry.second.second;
-
-    if (checkFaultMessages("SYSTEM", msg_val, key, value, b)) {
-      systemErrors[key].second = !systemErrors[key].second;
+    T1 value = entry.second.first;
+    bool exists = entry.second.second;
```

Comment:
I think exists is misleading. Can you think of a better name?

Reply:
second on naming

Reply:
I changed 'exists' to 'injected', here and in other functions.  Let me know (especially @lanseafood) if you think of a better name.

Reply:
I soon realized that 'injected' is not right, either.   I changed it to 'fault_in_progress".  Again, open to better name.

Reply:
How about **fault_active** and **faultActive**?

Reply:
You mean fault_active OR faultActive?  Yes I do like this better.

Reply:
Never mind, I see that both apply, in different places.